cpp/src/Slice/Scanner.cpp

```diff
@@ -1580,75 +1829,332 @@ YY_RULE_SETUP
+}
+	YY_BREAK
+/* ========== Identifiers and Keywords ========== */
+case 42:
+/* rule 42 can match eol */
+YY_RULE_SETUP
+#line 474 "src/Slice/Scanner.l"
+{
+    StringTokPtr ident = new StringTok;
+    ident->v = *yytext == '\\' ? yytext + 1 : yytext;
+    ident->v.erase(ident->v.find_first_of(" \t\v\n\r\f("));
+    *yylval = ident;
+    if(*yytext == '\\')
+    {
+        if(checkIdentifier(ident->v) == ICE_SCOPED_IDENTIFIER)
+        {
+            unit->error("Operation identifiers cannot be scoped: `" + (ident->v) + "'");
+        }
+        return ICE_IDENT_OPEN;
+    }
+    int st = checkKeyword(ident->v);
+    if(st == ICE_IDENTIFIER)
+    {
+        return ICE_IDENT_OPEN;
+    }
+    else if(st == ICE_SCOPED_IDENTIFIER)
+    {
+        unit->error("Operation identifiers cannot be scoped: `" + (ident->v) + "'");
+        return ICE_IDENT_OPEN;
+    }
+    else if(st == ICE_TAG)
+    {
+        return ICE_TAG_OPEN;
+    }
+    else if(st == ICE_OPTIONAL)
+    {
+        return ICE_OPTIONAL_OPEN;
+    }
+    else
+    {
+        return ICE_KEYWORD_OPEN;
+    }
+}
+	YY_BREAK
+case 43:
+YY_RULE_SETUP
+#line 511 "src/Slice/Scanner.l"
+{
+    StringTokPtr ident = new StringTok;
+    ident->v = *yytext == '\\' ? yytext + 1 : yytext;
+    *yylval = ident;
+    return *yytext == '\\' ? checkIdentifier(ident->v) : checkKeyword(ident->v);
+}
+	YY_BREAK
+/* ========== Whitespace ========== */
+/* Matches and consumes any whitespace, except for newlines. */
+case 44:
+YY_RULE_SETUP
+#line 521 "src/Slice/Scanner.l"
+{}
+	YY_BREAK
+/* Matches and consumes newlines, but only when the scanner isn't in a sub-scanner. */
+case 45:
+/* rule 45 can match eol */
+YY_RULE_SETUP
+#line 524 "src/Slice/Scanner.l"
+{
+    nextLine(yyleng);
+}
+	YY_BREAK
+/* ========== Others ========== */
+/* Matches and consumes a BOM, but only when the scanner has just started scanning a new file. */
+case 46:
+YY_RULE_SETUP
+#line 531 "src/Slice/Scanner.l"
+{}
+	YY_BREAK
+/* Matches invalid characters, one at a time to make this the 2nd lowest priority rule. All printable ASCII
+   * characters are valid (those between 32 and 127 inclusively), anything outside this range is invalid. */
+case 47:
+/* rule 47 can match eol */
+YY_RULE_SETUP
+#line 535 "src/Slice/Scanner.l"
+{
+    stringstream s;
+    s << "illegal input character: '\\";
+    s.width(3);
+    s.fill('0');
+    s << oct << static_cast<int>(static_cast<unsigned char>(yytext[0]));
+    s << "'";
+
+    unit->error(s.str());
+    return BAD_CHAR;
+}
+	YY_BREAK
+/* Matches any valid character (except newlines) not matched by another rule and fowards it to the parser.
+   * This is the lowest priority rule in the scanner, and is only active while not in a sub-scanner. */
+case 48:
+YY_RULE_SETUP
+#line 549 "src/Slice/Scanner.l"
+{
+    setLocation(yylloc);
+    return yytext[0];
+}
+	YY_BREAK
+case 49:
+YY_RULE_SETUP
+#line 554 "src/Slice/Scanner.l"
+YY_FATAL_ERROR( "flex scanner jammed" );
```

Comment:
Maybe VS can't see through the macro, but it really looks like `YY_FATAL_ERROR` is used in the generated code. Here, and about 15 other places.

And in the original changes, we set `#define YY_FATAL_ERROR fatalError`

Reply:
I will create an issue to fix this one in a separate PR.